wp/lib/bap_wp/src/constraint.ml

```diff
@@ -79,28 +90,60 @@ let format_registers (fmt : Format.formatter) (regs : reg_map) (jmp : Jmp.t)
     format_values fmt reg_vals;
     Format.fprintf fmt "\n%!"
 
-let format_path (fmt : Format.formatter) (p : path) (regs : reg_map)
-    (var_map : z3_expr Var.Map.t) : unit =
+(* Divide the path into the original path and the modified path. *)
+let partition_path (p : path) (regs : reg_map) ~orig:(_ : z3_expr Var.Map.t)
```

Comment:
This feels a bit hacky, but I assume there's no easy way to split based on `tid`s?

Reply:
I think it might be possible to split based on the `tid`s. One concern I do have is the case that the binaries have the same `tid` for a path. It's unlikely, but it may be possible.

Reply:
I wonder if the Right Thing to Do™ is to carry around the original source along with the constraints/path. But it's ok if we want to defer this work.

Reply:
I was able to pass in the subroutines for both the original and modified binaries to make the arguments more even, but I wonder if just passing in the original source was a better idea.